code_producers/src/llvm_elements/template.rs

```diff
@@ -149,8 +86,7 @@ impl<'a> TemplateCtx<'a> {
         create_load(producer, signals).into_pointer_value()
     }
 
-    /// Creates the necessary code to load a subcomponent counter given the expression used as id
-    pub fn load_subcmp_counter(
+    fn load_subcmp_counter(
         &self,
         producer: &dyn LLVMIRProducer<'a>,
         id: AnyValueEnum<'a>,
```

Comment:
Since in some circumstances we don't have the counter we could return an optional instead of returning always a pointer and checking if its a null pointer. Unless the null pointer is used for something else